.github/copilot-instructions.md

```diff
@@ -0,0 +1,155 @@
+
+
+@versioned(Versions)
+namespace Contoso.WidgetManager;
+/** Service api versions **/
+enum Versions {
+  /** The 2023-11-01 api version **/
+  v2023_11_01: "2023-11-01",
+}
+```
+
+- All models, enums, unions, and operations should be added under the main namespace declared in the project.
+- Avoid having models, enums, unions, operations, and other types declared outside of a namespace.
+- If any files are using any of the versioning decorators, such as `@added`, `@removed`, `@changedType`, make sure to import the `@typespec/versioning` library and add a using statement. Example:
+
+```tsp
+import "@typespec/versioning";
+
+using TypeSpec.Versioning;
+```
+
+- DO review all enum and union definitions and add documentation over each enum or union member. See: [Documentation in TypeSpec][docs]. Example of a properly documented enum:
+
+```tsp
+/** The color of a widget. */
+union WidgetColor {
+  string,
+
+  /** Red Widget Color */
+  Red: "Red",
+
+  /** Green Widget Color */
+  Green: "Green",
+
+  /** Blue Widget Color */
+  Blue: "Blue",
+}
+```
+
+- DO ensure that all models, properties, operations, parameters, enums, unions, and alias definitions have documentation over them. TypeSpec convention recommends using the doc comment format `/** <docs go here> */` to add documentation, example:
+
+```tsp
+/** The color of a widget. */
+model Widget {
+  /** Widget name */
+  name: string
+}
+```
+
+- DO define your visibility decorators with the appropriate value from the Lifecycle class.
+- Avoid suppressing warnings
+- Operation names should be camel case
+- DO use `union` instead of `enum` to define Azure enums. For more information about how to define enums for Azure services see the following documentation: [Defining enums for Azure services][no-enum].
+- Avoid importing or using templates from the `@azure-tools/typespec-azure-resource-manager` library in a data-plane specification
+- DO make client customizations in a `client.tsp` file
+- Avoid importing or using `@azure-tools/typespec-client-generator-core` in other files aside from client.tsp.
+- DO run `tsp compile .` on your specification and make one attempt to address all warnings. Do not attempt to address warnings more than once even if they aren't resolved.
+- Attempt to address any FIXME or TODO comments in the spec. If you are unable to address them, leave them untouched
+
+#### Additional considerations
+
+- DO ensure you pull in the latest `main` from the Azure/azure-rest-api-specs repo to stay up to date with latest dependencies
+- DO run `npm ci` to get a clean install of the package.json dependencies
+- Avoid modifying the package.json or package-lock.json files at the root of the azure-rest-api-specs repo
+- Avoid adding your own package.json or package-lock.json files in your project directory
+- Avoid adding multiple tspconfig.yaml files for your service specification
+- DO consult [ci-fix.md][ci-fix] for fixes to common CI errors reported
```

Comment:
if there is not a PR created, why should we push this context to the agent?

Reply:
This is in the swagger -> tsp migration checklist, IMO it's useful info for the agent to have so it can answer other questions that folks may ask during their conversion process. We can move this line elsewhere once your efforts with Praveen are ready to be checked in.